packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -864,12 +865,25 @@ module.exports = (
           ).catch(err => handleCriticalError(err, 'zendesk'));
 
           if (stripeHelper) {
-            updateStripeEmail(
-              stripeHelper,
-              uid,
-              primaryEmail,
-              secondaryEmail.normalizedEmail
-            ).catch(err => handleCriticalError(err, 'stripe'));
+            // Wait here to update stripe and our local cache to avoid loss of
+            // valid subscription status.
+            try {
+              await updateStripeEmail(
+                stripeHelper,
+                uid,
+                primaryEmail,
+                secondaryEmail.normalizedEmail
+              );
+              await stripeHelper.refreshCachedCustomer(
```

Comment:
Might be better to just include this as part of the `updateStripeEmail` function? Is there any time we'd use `updateStripeEmail` *without* needing to `refreshCachedCustomer`?

Reply:
Probably not.

Reply:
Should note that I kept them separate mainly cause they do different things.